### ISSUE TITLE
How to  adjust the parameters for train

### ISSUE THREAD

**Issue by skylning** [lightly-ai/lightly-train#558]
I train a instancesegement model as follow code:

import lightly_train

if __name__ == "__main__":
    lightly_train.train_instance_segmentation(
        out="out/myseg",    # Output directory
        model="dinov3/vitl16-eomt-inst-coco", 
        steps=90000,
        devices='auto',
        precision='bf16-mixed',
        accelerator='cuda',
        batch_size=4,
        num_workers=2,
        data={
            "path": "F:/label/rs/dataset_merged/yolo_format",      # Path to dataset directory
            "train": "images/train",    # Path to training images
            "val": "images/val",        # Path to validation images
            "names": {                  # Classes in the dataset                    
                0: "yz",        # Classes must match those in the annotation files
                1: "syz",
                2: "yl",
                # ...
            },
        },
    )

**my dataset has 3500 images and my GPU is a single RTX 4090 with 24G memory, and the train speed is very very slow , it spent 60 minutes and  the first step have not finish**

how adjust the parameters for train? someone can help me,thanks!!!

**Comment by stegmuel** [comment]
Hi,
By default the logging occurs every 100 steps (see [here](https://github.com/lightly-ai/lightly-train/blob/15fc97a65875862dde48375d8a395cda6eb682a9/src/lightly_train/_loggers/task_logger_args.py#L33)), so the first step probably has completed. 
That being said, this is still very slow. 
I would check the gpu utilization with `nvidia-smi` and if it is not high then set `num_workers='auto'`.
I would definitely reduce the number of steps, as 90k is what we used to train from the dinov3 initialization on COCO (significantly larger than 3.5k images). You are starting from an already finetuned model so you probably don't need that much. 

You can also reduce the image resolution via the `transform_args`:

```
import lightly_train


if __name__ == "__main__":
    lightly_train.train_instance_segmentation(
        out="out/my_experiment",
        model="dinov3/vitl16-eomt-inst-coco",
        data={
            # ...
        },
        transform_args={
            "image_size": (new_h, new_w),
             "val": { 
                 "image_size": (new_h, new_w), 
             }
        },
    )
```
Finally, you picked the largest model, you might want to use a smaller one.
For semantic segmentation we already support using larger patch size to improve speed, this will also be added for instance segmentation soon.

**Comment by skylning** [comment]
I change the model as dinov3/vitb16-eomt-inst-coco,the batch size=2, steps=9000,new coda as follow:
`import lightly_train

if __name__ == "__main__":
    lightly_train.train_instance_segmentation(
        out="out/myseg",    # Output directory
        model="dinov3/vitb16-eomt-inst-coco", 
        steps=9000,
        devices='auto',
        precision='bf16-mixed',
        accelerator='cuda',
        batch_size=2,
        num_workers="auto",
    
        data={
            "path": "F:/label/rs/dataset_merged/yolo_format",      # Path to dataset directory
            "train": "images/train",    # Path to training images
            "val": "images/val",        # Path to validation images
            "names": {                  # Classes in the dataset                    
                0: "yz",        # Classes must match those in the annotation files
                1: "syz",
                2: "yl",
                # ...
            },
        },
    )`

and the full log as:
...
Train Step 500/9000 | Train Loss: 14.0110
Train Step 600/9000 | Train Loss: 16.9525
Train Step 700/9000 | Train Loss: 9.4427
Train Step 800/9000 | Train Loss: 6.9345
...
  File "D:\lightly-train\src\lightly_train\_data\yolo_instance_segmentation_dataset.py", line 121, in __getitem__
    "binary_masks": binary_masks_np.astype(np.int_),
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
numpy._core._exceptions._ArrayMemoryError: Unable to allocate 1.66 GiB for an array with shape (544, 640, 640) and data type int64

**Comment by stegmuel** [comment]
I'll see if we can change the dtype for the binary masks to `bool` or `uint8`.
In the meantime, reducing the number of workers could help. 
Based on  the shape of  the mask this means that you can have up to 544 instances per image is this correct?

**Comment by stegmuel** [comment]
We changed the dtype of binary masks from int64 to uint8 in this #563. This should reduce the memory footprint of these masks by 8x and hopefully fix your issue. 
To benefit from this you would need to re-install from source as this is not released yet:
```
pip install "lightly-train @ git+https://github.com/lightly-ai/lightly-train.git"
```
